guardian/support-frontend
# Remove non-mandatory zip code validation message on checkout

## What are you doing in this PR?

Remove the validation on zip code used in two places: the legacy contributions checkout and the new generic checkout.



[**Trello Card**](https://trello.com/c/jpZYNi8q/933-non-mandatory-field-zip-code-incorrectly-labelled-as-mandatory-when-client-side-validation-errors-are-displayed)

## Why are you doing this?

On the old contributions and S+ checkouts, #6099 made the zip code field in personal details non-mandatory when the form is submitted. However the field still shows the validation error for zip code.

On the generic checkout the field was mandatory, which it shouldn't be.



## How to test

Go to any of these checkout flows and hit submit without entering any info - do not see error message beside zip code.

- US RC - Generic
- US S+ - Generic
- US RC - Non-generic
- US S+ - Non-generic

The following checkouts should have a mandatory validation of the post code/zip code in delivery/billing address

- GW
- Paper
- Tier Three - Generic



## Screenshots
### Non-generic

Before
![image](https://github.com/guardian/support-frontend/assets/114918544/64e2e2f4-f5d0-41f4-a114-28a4cb23a8af)


After
![image](https://github.com/guardian/support-frontend/assets/114918544/9a3f99bc-e352-486b-bce6-6d175914beb0)


### Generic

Before
![image](https://github.com/guardian/support-frontend/assets/114918544/5622efc1-14af-4e48-8656-d6d9d056ee04)


After
![image](https://github.com/guardian/support-frontend/assets/114918544/fa70da04-20c7-467f-b5a2-af49c4301468)
